securedrop_client/db.py

```diff
@@ -214,6 +220,61 @@ def __repr__(self) -> str:
         return '<Reply {}>'.format(self.filename)
 
 
+class DraftReply(Base):
+
+    __tablename__ = 'draftreplies'
+
+    id = Column(Integer, primary_key=True)
+    uuid = Column(String(36), unique=True, nullable=False)
+    timestamp = Column(DateTime, nullable=False)
+    source_id = Column(Integer, ForeignKey('sources.id'), nullable=False)
+    source = relationship("Source",
+                          backref=backref("draftreplies", order_by=id,
+                                          cascade="delete"))
+    journalist_id = Column(Integer, ForeignKey('users.id'))
+    journalist = relationship(
+        "User", backref=backref('draftreplies', order_by=id))
+
+    # Tracks where in this conversation the draft reply was sent.
+    # This points to the file_counter of the previous conversation item.
+    file_counter = Column(Integer, nullable=False)
+    content = Column(Text)
+
+    # This tracks the sending status of the reply.
+    send_status_id = Column(
+        Integer,
+        ForeignKey('replysendstatuses.id')
+    )
+    send_status = relationship("ReplySendStatus")
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    def __repr__(self) -> str:
+        return '<DraftReply {}>'.format(self.uuid)
+
+
+class ReplySendStatus(Base):
+
+    __tablename__ = 'replysendstatuses'
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String(36), unique=True, nullable=False)
+
+    def __init__(self, name: str) -> None:
+        super().__init__()
+        self.name = name
+
+    def __repr__(self) -> str:
+        return '<Reply status {}>'.format(self.name)
+
+
+class ReplySendStatusCodes(Enum):
+    """In progress (sending) replies can currently have the following statuses"""
```

Comment:
I see a `SAVED` status in the future :crystal_ball:

Reply:
haha yeah so at some point in this PR before I created the `DraftReply` object this status object was stored on the `Reply` and had three codes: `FAILED, SUCCESSFUL, PENDING`. but I removed `SUCCESSFUL` when I created the `DraftReply` as only drafts can be `FAILED` or `PENDING`, all replies are by construction `SUCCESSFUL`

I could imagine this is a place to store more detailed status codes about the failures in the future like `ENCRYPTION_FAILED_NO_SOURCE_KEY`, `SEND_FAILED_SOURCE_DELETED`. We could also imagine some more granular pending statuses like `PENDING_ENCRYPTION_IN_PROGRESS`, `PENDING_AWAITING_SERVER_RESPONSE`.